### ISSUE TITLE
Test: VS Code with Orca screen reader

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#91063]
Refs: https://github.com/microsoft/vscode/issues/90446

- [x] Linux @joaomoreno
- [x] Linux @rebornix

Complexity: 4

We are now supporting Orca screen reader on Linux. We should do some exploratory testing on how Orca behaves with VS Code. File issues to me @isidorn and then I will forward them to the corresponding owner.

1. Configure Orca to use the latest version by following these [instructions](https://github.com/microsoft/vscode/issues/90446#issuecomment-585782600)
2. Turn on Orca by Universal Access > Screen Reader > On. [Docs](https://help.ubuntu.com/stable/ubuntu-help/a11y-screen-reader.html.en)
2. Set `"editor.accessibilitySupport": "on"` in vscode insiders settings
3. Restart VS Code

Verify the following:
* Navigation character by character in the Editor correctly reads the characters
* Navigation word by word in the Editor correctly reads the words
* Suggest widget is accessible
* Take some list / tree (example Explorer, Extensions view) and verify navigating through it the content is nicely read
* Test out the main menu is nicely read out and can be navigated via keyboard
* Try out some other area of your liking and try if it's accessibility with Orca

**Comment by rebornix** [comment]
Since there is no sound support for Ubuntu vm in HyperV, I set up a new Ubuntu 18 vm in Parallel Desktop and installed following packages to get `orca` working

```bash
# git
sudo apt install git
git clone https://gitlab.gnome.org/GNOME/orca.git
cd orca/

# packages for building orca
sudo apt install automake
sudo apt install autoconf
sudo apt install autopoint
sudo apt install yelp-tools
sudo apt install -y python-gi-dev
sudo apt-get install libatspi2.0-dev
sudo apt-get install libatk-bridge2.0

# sound support https://gstreamer.freedesktop.org/documentation/installing/on-linux.html?gi-language=c
sudo apt-get install libgstreamer1.0-0 gstreamer1.0-plugins-base gstreamer1.0-plugins-good gstreamer1.0-plugins-bad gstreamer1.0-plugins-ugly gstreamer1.0-libav gstreamer1.0-doc gstreamer1.0-tools gstreamer1.0-x gstreamer1.0-alsa gstreamer1.0-gl gstreamer1.0-gtk3 gstreamer1.0-qt5 gstreamer1.0-pulseaudio


# build
make
sudo make install
```

**Comment by isidorn** [comment]
@rebornix thanks for pasting the instructions here

**Comment by joaomoreno** [comment]
Doesn't seem to work in Ubuntu 19.10.

**Comment by isidorn** [comment]
Our blind users seem to use Arch Linux.
@rebornix let me know if things seem to work for you with Ubuntu 18

**Comment by joaomoreno** [comment]
I couldn't get Orca to work at all in Arch, will try again on my home setup...

I couldn't get it to work on Ubuntu 19.10, will still try 18.04.

**Comment by rebornix** [comment]
It works fine on Ubuntu 18.04 in PD based on my testing. Ran into a couple of issues but in overall it's good to me.

**Comment by joaomoreno** [comment]
Failed to get any screen reading on Arch Linux.

**Comment by jvesouza** [comment]
> Failed to get any screen reading on Arch Linux, even on other apps. Orca simply does not speak. I've tried the stable Arch release as well as orca-git.

Some questions:
Did you install speech-dispatcher?
Speech-dispatcher is running?
What synth did yu install?

**Comment by deepak1556** [comment]
Overall works well, will continue playing with it.

**Comment by joaomoreno** [comment]
Thanks @deepak1556.

I had speech-dispatcher. I verified it was running, PulseAudio was showing those playback sources. Orca was running and the settings seemed fine. Firefox was automatically scrolling down, as Orca does it. But simply no voice came back.

I have removed everything since.

**Comment by isidorn** [comment]
Thanks all a lot for testing!

**Comment by jvesouza** [comment]
@joaomoreno It is possible to run some tests if time permits.

* Check if the synthesizer is working by executing the following command:
> espeak 'hello world'

* Check if the speech-dispatcher is working by executing the following command:
> spd-say 'hello world'

* Change the synthesizer. You can try espeak or espeak-ng.
In my arch environment I already had problems when I used espeak-ng.

* Run orca in the terminal to check for any error messages.
> orca -r

* Run orca in debug mode to check for possible errors.
> orca -r --debug-file=orca.debug.out

**Comment by joaomoreno** [comment]
@jvesouza The hint to use `espeak-ng` worked: now everything seems to work fine with that! Thanks a lot!

**Comment by joaomoreno** [comment]
@isidorn Now orca works in GTK apps (Firefox, Files), but I still get nothing from Code. :think

**Comment by jvesouza** [comment]
@joaomoreno Do you have the "editor.accessibilitySupport": "on" on your settings?
Try to pass the flag  --force-renderer-accessibility when executing vscode.
If I understand correctly, when editor.accessibilitySupport is enabled and the environment is linux, the flag is automatically passed to electron.

**Comment by isidorn** [comment]
@joaomoreno weird. I would try what @jvesouza is suggesting. However if you do not have anything read also when navigating trees than it is not that.
